binary_search_tree/tree.py

```diff
@@ -14,46 +14,136 @@ class Tree:
     def __init__(self):
         self.root = None
 
-    # Time Complexity: 
-    # Space Complexity: 
+    # Time Complexity: 0(log(n)
+    # Space Complexity: 0(1)
     def add(self, key, value = None):
-        pass
+        added = TreeNode(key, value)
+        if not self.root:
+            self.root = added
+
+        parent = self.root
+        while True:
```

Comment:
We generally try to avoid while loops whose conditions can never be Falsey. With while loops, we always want the body of the while loop to be making progress toward the condition being False. 

How might you refactor you code to eliminate this? Hint: Look at your `find` function